src/natcap/invest/urban_mental_health.py

```diff
@@ -568,229 +602,289 @@ def execute(args):
 
     LOGGER.info("Start preprocessing")
 
+    # get target pixel size for outputs
+    if args['scenario'] in ['tcc_ndvi', 'ndvi']:
+        pixel_size = pygeoprocessing.get_raster_info(
+            args['ndvi_base'])['pixel_size']
+    else:
+        pixel_size = pygeoprocessing.get_raster_info(
+            args['lulc_base'])['pixel_size']
+
+    pixel_radius = int(round(args['search_radius']/pixel_size[0]))
+    LOGGER.info(f"Search radius {args['search_radius']} results in "
+                f"buffer of {pixel_radius} pixels")
+    if pixel_radius == 0:
+        raise ValueError(
+            f"Search radius {args['search_radius']} yielded pixel_radius "
+            "of zero. Please increase search radius.")
+
+    # Users should input the AOI to which they want outputs clipped
+    # InVEST will take care of buffering the processing AOI to ensure
+    # correct edge pixel calculation
+    aoi_info = pygeoprocessing.get_vector_info(args['aoi_path'])
+    aoi_projection = aoi_info["projection_wkt"]
+    aoi_sr = osr.SpatialReference()
+    aoi_sr.ImportFromWkt(aoi_projection)
+    aoi_bbox = aoi_info["bounding_box"]
+
+    # Expand target bounding box to ensure correct edge pixel calculation
+    aoi_buffered_bbox = aoi_bbox + numpy.array(
+        [-args['search_radius'], -args['search_radius'],
+            args['search_radius'], args['search_radius']])
+    aoi_buffered_bbox = list(aoi_buffered_bbox)
+
+    raster_to_method_dict = {'ndvi_base': 'cubic', 'ndvi_alt': 'cubic',
+                             'lulc_base': 'near', 'lulc_alt': 'near'}
+    input_align_list = []
+    output_align_list = []
+    resample_method_list = []
+    for input_raster, resample_method in raster_to_method_dict.items():
+        if args[input_raster]:
+            input_align_list.append(args[input_raster])
+            output_align_list.append(file_registry[input_raster+'_aligned'])
```

Comment:
We've generally adopted f-strings for string concatenation. I think it's technically "faster" and can be more readable than using the plus operator.